lib/keyrod/fedcloud_client.rb

```diff
@@ -0,0 +1,122 @@
+require 'faraday'
+require 'json'
+require 'net/http'
+
+module Keyrod
+  class FedcloudClient
+    attr_reader :site, :access_token, :auth_path, :ssl
+
+    PROJECTS_PATH = '/v3/auth/projects'.freeze
+    SCOPED_PATH = '/v3/auth/tokens'.freeze
+    REDIRECT_HEADER = 'WWW-Authenticate'.freeze
+    REDIRECT_REGEXP = /(?<=\')(.*?)(?=\')/
```

Comment:
This regular expression is unnecessary complicated. I think we can always expect that the header will start with `Keystone uri=` (non-casesensitive). But lookaheads/lookbehinds are overkill anyway.